test/clojure/core_test/assoc_bang.cljc

```diff
@@ -0,0 +1,65 @@
+(ns clojure.core-test.assoc-bang
+  (:require [clojure.test :refer [deftest testing are]]
+            [clojure.core-test.portability #?(:cljs :refer-macros :default :refer) [when-var-exists]]))
+
+(when-var-exists assoc!
+
+  (deftest test-assoc!
+
+    (testing "maps"
+      (testing "maps - single value"
+        (are [expected map key val] (= expected (persistent! (assoc! (transient map) key val)))
+                                    {:a 1} {} :a 1
+                                    {:a 1 :b 2} {:a 1} :b 2
+                                    {:a 3 :b 2} {:a 1 :b 2} :a 3))
+      (testing "maps - multiple values"
+        (are [expected map kvs] (= expected (persistent! (apply assoc! (transient map) kvs)))
+                                {:a 1 :b 2} {} [:a 1 :b 2]
+                                {:a 1 :b 3} {} [:a 1 :b 2 :b 3]
+                                {:a 1 :b 3 :c 5 :d 7} {:a 1 :b 2} [:b 3 :c 5 :d 7])))
+
+    (testing "vectors"
+      (testing "vectors - single value"
+        (are [expected vec index val] (= expected (persistent! (assoc! (transient vec) index val)))
+                                      [nil] [] 0 nil
+                                      [0] [] 0 0
+                                      [0 3 2] [0 1 2] 1 3
+                                      [0 1 2 3] [0 1 2] 3 3))
+      (testing "vectors - multiple values"
+        ; assoc coll index-1 val-1 index-2 val-2 ...
+        (are [expected vec ivs] (= expected (persistent! (apply assoc! (transient vec) ivs)))
```

Comment:
Same suggestion as before that naming the binding `idxs` would be more idiomatic than `ivs`

Reply:
I'm sorry, I still don't understand, `ivs` are not indices, they are indices and values (`[index-1 value-1 index-2 value-2 ...]`) 🤔